icu4c/source/i18n/unitconverter.cpp

```diff
@@ -107,6 +113,36 @@ struct Factor {
 
         factorNum *= siApplied;
     }
+
+    void substituteConstants() {
+        double constantsValues[CONSTANTS_COUNT];
+
+        constantsValues[CONSTANT_FT2M] = 0.3048;
+        constantsValues[CONSTANT_PI] = 411557987.0 / 131002976.0;
+        constantsValues[CONSTANT_GRAVITY] = 9.80665;
+        constantsValues[CONSTANT_G] = 6.67408E-11;
+        constantsValues[CONSTANT_LB2KG] = 0.45359237;
+        constantsValues[CONSTANT_GAL_IMP2M3] = 0.00454609;
+
+        for (int i = 0; i < CONSTANTS_COUNT; i++) {
+            if (this->constants[i] == 0) continue;
+
+            substituteSingleConstant(this->constants[i], constantsValues[i]);
+            this->constants[i] = 0;
+        }
+    }
+
+    void substituteSingleConstant(int32_t constantPower,
```

Comment:
Optional: I would combine these two methods, since this method is only 5 lines long.